File: java/datastructure/datastructure/Tree/BinaryTreeNode.java
package datastructure.Tree;

public class BinaryTreeNode<T> {

    private BinaryTreeNode<T> left; 
    private BinaryTreeNode<T> right; 
    private T data; 
    
    public BinaryTreeNode() {
        this.left = null; 
        this.right = null; 
    }

    public BinaryTreeNode(T ele) {
        this.data = ele; 
        this.left = null; 
        this.right = null; 
    }

    public void setLeft(BinaryTreeNode<T> left) {
        this.left = left; 
    }

    public void setRight(BinaryTreeNode<T> right) {
        this.right = right; 
    }

    public BinaryTreeNode<T> getLeft() {
        return this.left; 
    }

    public BinaryTreeNode<T> getRight() {
        return this.right; 
    }

    public void setData(T data) {
        this.data = data; 
    }

    public T getData() {
        return this.data; 
    }
    
}

File: cpp/datastructure/datastructure/Tree/BinaryTreeNode.cpp
#ifndef BINARYTREENODE_H 
#define BINARYTREENODE_H 

template <typename T> 
class BinaryTreeNode {
    public: 
        BinaryTreeNode(); 
        void getData() const; 
        void setData(T data); 
        BinaryTreeNode<T> *getLeft() const; 
        BinaryTreeNode<T> *getRight() const; 
        void setLeft(BinaryTreeNode<T> *new_left); 
        void setRight(BinaryTreeNode<T> *new_right); 
    private: 
        T data; 
        BinaryTreeNode *left; 
        BinaryTreeNode *right; 
}; 

template <typename T> 
BinaryTreeNode<T>::BinaryTreeNode() {
    this->left = None; 
    this->right = None; 
}

template <typename T> 
void BinaryTreeNode<T>::getData() const {
    return this->data; 
}

template <typename T> 
void BinaryTreeNode<T>::setData(T data) {
    this->data = data; 
}



#endif 